### PR TITLE
BC-5513 - new folder structure

### PR DESCRIPTION
**Pull request by odalys-dataport**
# Short Description

To avoid folder clutter, we want to add one folder for each type of module (page, feature, ui, data and util) instead of the prefix.

https://documentation.dbildungscloud.dev/docs/nuxt-client/ProjectStructure

## Links to Ticket and related Pull-Requests

BC-5513

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/modules/data/board/BoardFocusHandler.composable.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
 } from "@vueuse/core";
 import { computed, nextTick, onMounted, onUnmounted, ref, Ref } from "vue";
 // WIP: InlineEditInteractionHandler needs to be extracted into it's own module... :: Feature-Modules have to be imported using the pattern '@feature-<name>'eslint@typescript-eslint/no-restricted-imports
-import { useInlineEditInteractionHandler } from "../feature-board/shared/InlineEditInteractionHandler.composable";
+import { useInlineEditInteractionHandler } from "../../feature/board/shared/InlineEditInteractionHandler.composable";
```
Comment (uidp) [inline src/modules/data/board/BoardFocusHandler.composable.ts:13]: Shouldn't that be `@feature-board`?

File: src/modules/feature/board-file-element/content/FileContent.vue [hunk 2]
```diff
@@ -71,3 +71,5 @@ export default defineComponent({
 	},
 });
 </script>
+./display/FileDisplay.vue./inputs/FileInputs.vue
+./display/FileDisplay.vue./inputs/FileInputs.vue
```
Comment (uidp) [inline src/modules/feature/board-file-element/content/FileContent.vue:75]: What is this about?

File: src/modules/feature/board-link-element/composables/PreviewGenerator.composable.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import {
 	convertDownloadToPreviewUrl,
 	isPreviewPossible,
 } from "@/utils/fileHelper";
-import { useFileStorageApi } from "@feature-board-file-element";
+import { useFileStorageApi } from "@/modules/feature/board-file-element";
```
Comment (uidp) [inline src/modules/feature/board-link-element/composables/PreviewGenerator.composable.ts:6]: Should be `@feature-board-file-element`?

File: src/modules/feature/board-text-element/RichTextContentElementDisplay.unit.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { RenderHTML } from "@feature-render-html";
+import { RenderHTML } from "@/modules/feature/render-html";
```
Comment (uidp) [inline src/modules/feature/board-text-element/RichTextContentElementDisplay.unit.ts:1]: Should be `@feature-render-html`?

File: src/modules/feature/board-text-element/RichTextContentElementDisplay.vue [hunk 5]
```diff
@@ -41,3 +41,4 @@ export default defineComponent({
 	margin-top: 0;
 }
 </style>
+@/modules/feature/feature-render-html @/modules/feature/render-html
```
Comment (uidp) [inline src/modules/feature/board-text-element/RichTextContentElementDisplay.vue:44]: What is this about?

File: src/modules/feature/board/card/ContentElementList.vue [hunk 6]
```diff
@@ -267,3 +267,7 @@ export default defineComponent({
 	},
 });
 </script>
+@/modules/feature/feature-board-external-tool-element@/modules/feature/feature-board-file-element@/modules/feature/feature-board-link-element@/modules/feature/feature-board-submission-element@/modules/feature/feature-board-text-element@/modules/feature/feature-board-drawing-element
+@/modules/feature/board-external-tool-element
+@/modules/feature/board-file-element @/modules/feature/board-submission-element
+@/modules/feature/board-text-element
```
Comment (uidp) [inline src/modules/feature/board/card/ContentElementList.vue:273]: What is this about?

File: src/modules/ui/board/BoardMenuActionDelete.unit.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
-import { BoardMenuScope } from "@/components/ui-board/board-menu-scope";
-import { MENU_SCOPE } from "@/components/ui-board/injection-tokens";
+import { BoardMenuScope } from "@/modules/ui/board/board-menu-scope";
```
Comment (uidp) [inline src/modules/ui/board/BoardMenuActionDelete.unit.ts:1]: Shouldn't that be `@ui-board`? That would additionally require `BoardMenuScope` to be published in the barrel file `src/modules/ui/board/index.ts`.

File: src/modules/ui/board/BoardMenuActionDelete.unit.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
...
+import { MENU_SCOPE } from "@/modules/ui/board/injection-tokens";
```
Comment (uidp) [inline src/modules/ui/board/BoardMenuActionDelete.unit.ts:2]: `@ui-board` (see above).

File: src/modules/ui/board/LineClamp.unit.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import { mount } from "@vue/test-utils";
-import LineClamp from "@/components/ui-board/LineClamp.vue";
+import LineClamp from "@/modules/ui/board/LineClamp.vue";
```
Comment (uidp) [inline src/modules/ui/board/LineClamp.unit.ts:2]: `from "./LineClamp.vue"`? We can use the relative import here.

File: src/modules/util/device-detection/TouchDetection.composable.unit.ts [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import { useTouchDetection } from "@/components/util-device-detection/TouchDetection.composable";
+import { useTouchDetection } from "@/modules/util/device-detection/TouchDetection.composable";
```
Comment (uidp) [inline src/modules/util/device-detection/TouchDetection.composable.unit.ts:1]: Should use the relative import `from "./TouchDetection.composable"`

File: src/modules/util/device-detection/util/isTouchEvent.unit.ts [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import { isTouchEvent } from "@/components/util-device-detection/util/isTouchEvent";
+import { isTouchEvent } from "@/modules/util/device-detection/util/isTouchEvent";
```
Comment (uidp) [inline src/modules/util/device-detection/util/isTouchEvent.unit.ts:1]: Should use the relative import `from "./isTouchEvent"`.

File: src/modules/util/input-masks/InputMask.unit.ts [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import { createInputMask } from "@/components/util-input-masks/InputMask.factory";
+import { createInputMask } from "@/modules/util/input-masks/InputMask.factory";
```
Comment (uidp) [inline src/modules/util/input-masks/InputMask.unit.ts:1]: Should use the relative import `from ./InputMask.factory`.

File: tests/test-utils/api-mocks/fileStorageApiMock.ts [hunk 13]
```diff
@@ -1,5 +1,5 @@
 import { FileRecordResponse } from "@/fileStorageApi/v3";
-import { useFileStorageApi } from "@feature-board-file-element";
+import { useFileStorageApi } from "@/modules/feature/board-file-element";
```
Comment (uidp) [inline tests/test-utils/api-mocks/fileStorageApiMock.ts:2]: Should be `@feature-board-file-element`?

File: tests/test-utils/factory/provisioningOptionsDataFactory.ts [hunk 14]
```diff
@@ -1,5 +1,5 @@
 import { Factory } from "fishery";
-import { ProvisioningOptions } from "@/components/data-provisioning-options";
+import { ProvisioningOptions } from "@/modules/data/provisioning-options";
```
Comment (uidp) [inline tests/test-utils/factory/provisioningOptionsDataFactory.ts:2]: Should be `@data-provisioning-options`?

File: tests/test-utils/composable-mocks/index.ts [hunk 15]
```diff
@@ -1 +1 @@
-export * from "../../../src/components/feature-board/test-utils/sharedElementTypeSelectionMock";
+export * from "../../../src/modules/feature/board/test-utils/sharedElementTypeSelectionMock";
```
Comment (uidp) [inline tests/test-utils/composable-mocks/index.ts:1]: Should be `@feature-board`? Would require `sharedElementTypeSelectionMock` to be published from the barrel file `src/modules/feature/board/index.ts`

**Review by uidp — COMMENTED** [review]
We definitely have to centralise the module paths in one single file, e.g. the typescript config. And then reference them from all the other configs. Not that we have to write them in several places.

File: src/modules/data/board/BoardFocusHandler.composable.ts (hunk 1 above)
Comment (odalys-dataport) [reply src/modules/data/board/BoardFocusHandler.composable.ts:13]: Probably, the comment above has been existing for a while now, there are some refactors planned for this, that's why I didn't touch this.

File: src/modules/feature/board-file-element/content/FileContent.vue (hunk 2 above)
Comment (odalys-dataport) [reply src/modules/feature/board-file-element/content/FileContent.vue:75]: Apparently auto renaming imports after moving files doesn't work so well in vs code 😄

File: src/modules/ui/board/BoardMenuActionDelete.unit.ts (hunk 7 above)
Comment (odalys-dataport) [reply src/modules/ui/board/BoardMenuActionDelete.unit.ts:1]: I'm gonna guess that they weren't added on purpose to the barrel file and import them with `./...`

**Review by uidp — APPROVED** [review]

**Pull request merged**